custom/icds_reports/data_pull/queries.py

```diff
@@ -0,0 +1,120 @@
+from io import StringIO
+
+from django.db import connections
+from django.utils.dateparse import parse_date
+from django.utils.functional import cached_property
+
+from dateutil.relativedelta import relativedelta
+
+
+class BaseQuery:
+    query_file_path = ""
+
+    def run(self, db_alias):
+        string_buffer = StringIO()
```

Comment:
any memory concerns here? how big are the result sets?

Reply:
These files are usually in few MBs, definitely less than 20 MBs in case of big files.